Fix structure of search page's filters

Stef-Rousset opened the pull request:
#### :tophat: What? Why?
This PR adds structure to the search page's filters, by replacing div tags with unordered list ul li tags.
This PR is issued from the audit of Angers city (page 53), and corresponds to criteria 1.3.1 of WCAG.

#### Testing
As a user, perform a search on any page with the search input located on the main_bar of the page.
On the search page results, open your inspector and see that the filters on the left are displayed in ul/li structured list.

### :camera: Screenshots
![Description](URL)
<img width="921" alt="Capture d’écran 2025-06-02 à 18 05 30" src="https://github.com/user-attachments/assets/eeeb24a3-1799-4e0f-8722-51703b89275d" />

:hearts: Thank you!

alecslupu reviewed (changes requested):
while checking this PR, i have noticed 2 things: 
1. The inner elements (Process, Assemblies etc) are being rendered in divs 
2. I have seen that Filter is clickable, but the list does not hide as the interface suggests it. 
Could you, please , handle those?

alecslupu reviewed (changes requested):
I think there is a small issue here, the markup doesn't seem to be generated as it should be : 

- ul 
  - li
    -  ul 
      - li
      - li
      - li
  - li
but is generated as: 
- ul 
  - li 
    - ul 
      - li
    - ul 
	  - li 
    - ul 
	  - li 
  - li 

The arrow works only with click, when using keyboard is not working. so we need to some key down events.

Stef-Rousset commented:
hello @alecslupu , thanks for your review,  I have updated the ul/li structure and added the hadle of arrow when navigating with keyboard. Please let me know if it is ok now !

alecslupu commented on decidim-core/app/views/decidim/searches/_filters.html.erb:
```diff
@@ -16,48 +16,68 @@
+      </li>
       <% @blocks.each do |elements| %>
-        <div>
+        <li>
```
Fixed by [697f48b](https://github.com/decidim/decidim/pull/14789/commits/697f48b05865125c3ac55c1d4a0ca50ca7b132b5)
